pandas/tests/api/test_api.py

```diff
@@ -245,9 +252,10 @@ def test_datetime():
     from datetime import datetime
     import warnings
 
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore", FutureWarning)
-        assert datetime(2015, 1, 2, 0, 0) == pd.datetime(2015, 1, 2, 0, 0)
+    if compat.PY37:
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", FutureWarning)
+            assert datetime(2015, 1, 2, 0, 0) == pd.datetime(2015, 1, 2, 0, 0)
```

Comment:
Why only test this for python 3.7? The equality should also hold for python 3.6 ?

Reply:
See my analysis above.  I couldn't figure out how to make `pd.datetime` have a deprecation message when doing `pd.datetime(2015,1,2,0,0)` and also support `pd.datetime.now()` with a deprecation message when using python 3.6.

Reply:
But this test should also pass when there is no warning? (the test doesn't assert there is a warning, only that the functionality is still working)

Reply:
The issue with python 3.6 and using `pd.datetime(2015,1,2,0,0)` is that the implementation in this PR throws an exception.  Current master just succeeds without any warning at all.

Reply:
I think a correct result is more important right now than having the deprecation warning in all cases.